main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -368,12 +377,24 @@ object Type {
     * Constructs the tuple type (A, B, ...) where the types are drawn from the list `ts`.
     */
   def mkTuple(ts: List[Type]): Type = {
+    if (ts.lengthIs <= 1) {
+      throw InternalCompilerException(s"Illegal tuple argument list length: $ts")
+    }
     val init = Type.Cst(TypeConstructor.Tuple(ts.length))
     ts.foldLeft(init: Type) {
       case (acc, x) => Apply(acc, x)
     }
   }
 
+  /**
+    * Constructs a Unit, Singleton Type, or Tuple from the given types.
+    */
+  def mkUnitOrTuple(ts: List[Type]): Type = ts match {
+    case Nil => Type.Unit
+    case List(elem) => elem
+    case _ => mkTuple(ts)
+  }
+
```

Comment:
To the extend possible, I prefer to have these helper `mkSomeType` functions inside `Typer`.

Reply:
By that I mean, if they are only used in Typer they should be there.